### PR TITLE
largest-series-product: add test template

### LINKED ISSUE #1957: largest-series-product: add test template
## Task
Create a test template file for `largest-series-product`

### Details
1. If this is your first time contributing to this repository, welcome!
    - Contributing to a repository on GitHub requires a [fork](https://help.github.com/en/articles/fork-a-repo)
1. Create a branch on your local repository called `largest-series-product-add-test-template`
    - Run: `git checkout -b largest-series-product-add-test-template`
1. Clone problem-specifications repository; this is used to populate the template
    - Run: `git clone https://github.com/exercism/problem-specifications ../problem-specifications`
1. Create `exercises/largest-series-product/.meta/template.j2`
    - See [Generator Documentation] for more information
1. Generate test file from new template
    - Run: `bin/generate_tests.py largest-series-product`

At this point, if the generator fails, or you are otherwise stuck,
please feel free to reach out! It is perfectly acceptable to submit
an incomplete PR in order to get feedback or assistance.

### Resources
- [Contributing Guide]
- [Generator Documentation]
- [Closing issues using keywords]

[Contributing Guide]: https://github.com/exercism/python/blob/master/docs/CONTRIBUTING.md
[Generator Documentation]: https://github.com/exercism/python/blob/master/docs/GENERATOR.md
[Closing issues using keywords]: https://help.github.com/en/articles/closing-issues-using-keywords

### PR DESCRIPTION
**Pull request by MathWhiz**
Fixes #1957

### CODE REVIEW

**Comment by MathWhiz** [comment]
It appears that the expected function is largestProduct (according to the canonical data), but this track defines largest_product. Which should be used?

File: exercises/largest-series-product/.meta/template.j2 [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{{ macros.header() }}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases -%}
+    def test_{{ case["description"] | to_snake }}(self):
+        {% set input = case["input"] -%}
+        {% set digits = input["digits"] -%}
+        {% set span = input["span"] -%}
+        {% set expected = case["expected"] -%}
+        {% set function = case["property"] -%}
```
Comment (cmccandless) [inline exercises/largest-series-product/.meta/template.j2]: This should address the naming discrepancy.

```suggestion
        {% set function = to_snake(case["property"]) -%}
```

**Comment by cmccandless** [comment]
@MathWhiz You'll need to regenerate the test file. This must be done whenever the template is changed.

**Comment by yawpitch** [comment]
Renaming this so we can easily see what exercise it relates to.

**Comment by MathWhiz** [comment]
I get an error trying to generate the file now :\

**Comment by cmccandless** [comment]
@MathWhiz Can you paste the error here?

**Comment by MathWhiz** [comment]
It appears that it isn't `to_snake(case["property"])` but `case["property"] | to_snake`

**Comment by cmccandless** [comment]
@yawpitch how do we feel about that "extra-credit" test case?

**Comment by yawpitch** [comment]
> @yawpitch how do we feel about that "extra-credit" test case?

@cmccandless personally I think it's a bit of a shame to lose the "extra-credit" case ... there's _lots_ of ways of implementing this such that you get exceptionally poor performance, and it would at least help the student see that if they attempt the extra credit.

That said I do think hints.md should at least mention any extra credit tests.

**Comment by cmccandless** [comment]
> personally I think it's a bit of a shame to lose the "extra-credit" case ... there's lots of ways of implementing this such that you get exceptionally poor performance, and it would at least help the student see that if they attempt the extra credit.

I agree.

@MathWhiz , please see the [docs](https://github.com/exercism/python/blob/master/docs/GENERATOR.md#conventions) on adding track-specific tests, and include the removed extra credit case.

**Comment by yawpitch** [comment]
Looks good to me. Shame that `black` can't really wrap 80 character strings like that ... while I'm sure there's a way to use `textwrap` to do that _before_ the `black` format pass in a way that would be preserved I very much doubt this single test case is worth the effort.

**Pull request merged**